runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6086,6 +6086,12 @@ TR_J9ByteCodeIlGenerator::loadArrayElement(TR::DataType dataType, TR::ILOpCodes
       return;
       }
 
+   loadArrayElementWithoutHelper(dataType, nodeop, checks);
+   }
+
+void
+TR_J9ByteCodeIlGenerator::loadArrayElementWithoutHelper(TR::DataType dataType, TR::ILOpCodes nodeop, bool checks)
```

Comment:
I think the changes look correct.  One suggestion I have is to keep the single `loadArrayElement` method, and instead add an extra `bool isNotFlattened` or `isNotValueType` argument, which would be `false` by default and `true` in the case of `loadInvokeCacheArrayElements`.  That would make the treatment similar to the treatment for `checks`.

Reply:
Updated in  5e0c88e